src/tests/functional/plugin/shared/include/behavior/plugin/ov_auto_batching_tests.hpp

```diff
@@ -0,0 +1,165 @@
+    size_t num_requests;
+    size_t num_batch;
+    std::vector<std::shared_ptr<ov::Model>> fn_ptrs;
+
+    void TestAutoBatch() {
+        auto core = ov::test::utils::PluginCache::get().core();
+
+        ov::OutputVector outputs;
+        std::vector<std::pair<std::shared_ptr<ov::Model>, ov::InferRequest>> irs;
+        std::vector<ov::InferRequest> irs_ref;
+        std::vector<int> outElementsCount;
+
+        for (size_t i = 0; i < fn_ptrs.size(); ++i) {
+            auto model = fn_ptrs[i];
+            auto inputs = model->inputs();
+            for (auto n : inputs) {
+                n.get_node()->set_output_type(0, ov::element::f32, n.get_shape());
+            }
+            ov::AnyMap config;
+            if (target_device.find("GPU") != std::string::npos) {
+                config[CONFIG_KEY(GPU_THROUGHPUT_STREAMS)] = std::to_string(num_streams);
+                config["INFERENCE_PRECISION_HINT"] = "f32";
+            }
+
+            if (target_device.find("CPU") != std::string::npos) {
+                config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = std::to_string(num_streams);
+                config[CONFIG_KEY(ENFORCE_BF16)] = CONFIG_VALUE(NO);
+            }
+            // minimize timeout to reduce test time
+            config[CONFIG_KEY(AUTO_BATCH_TIMEOUT)] = std::to_string(1);
+
+            auto compiled_model = core->compile_model(model, std::string(ov::test::utils::DEVICE_BATCH) + ":" +
+                                                      target_device + "(" + std::to_string(num_batch) + ")",
+                                                      config);
+
+            auto network_outputs = model->outputs();
+            ASSERT_EQ(network_outputs.size(), 1) << " Auto-Batching tests use networks with single output";
+            auto output = *network_outputs.begin();
+            for (size_t j = 0; j < num_requests; j++) {
+                outputs.push_back(output);
+                outElementsCount.push_back(
+                        std::accumulate(begin(fn_ptrs[i]->get_output_shape(0)), end(fn_ptrs[i]->get_output_shape(0)), 1,
+                                        std::multiplies<size_t>()));
+
+                auto inf_req = compiled_model.create_infer_request();
+                irs.push_back({model, inf_req});
+
+                auto compiled_model_ref = core->compile_model(model, ov::test::utils::DEVICE_TEMPLATE);
+                auto inf_req_ref = compiled_model_ref.create_infer_request();
+                irs_ref.push_back(inf_req_ref);
+
+                std::vector<ov::Tensor> inData;
+                for (auto input : inputs) {
+                    auto tensor = ov::test::utils::create_and_fill_tensor(input.get_element_type(), input.get_shape());
+                    if (use_get_tensor)
+                        memcpy(reinterpret_cast<void *>(inf_req.get_tensor(input).data()),
+                               reinterpret_cast<const void *>(tensor.data()), tensor.get_byte_size());
```

Comment:
```suggestion
                        memcpy(inf_req.get_tensor(input).data(), tensor.data(), tensor.get_byte_size());
```
Casts shouldn't be required as tensor `data()` returns `void*`

Reply:
fixed